patch-diff-report-tool/src/main/java/com/github/checkstyle/data/Statistics.java

```diff
@@ -88,27 +98,57 @@
      */
     private int uniqueMessagesPatch;
 
+    /**
+     * Getter for number of records per severity for difference.
+     *
+     * @return number of records per severity.
+     */
     public final Map<String, Integer> getSeverityNumDiff() {
+        final Map<String, Integer> severityNumDiff = new HashMap<>(severityNumRemoved);
+        severityNumAdded.forEach((key, value) -> severityNumDiff.merge(key, value, Math::addExact));
```

Comment:
`Math::addExact` will throw an error if there is an overflow. I am not sure we really want this. There are alot more instances of this.

@romani ping.

Reply:
https://docs.oracle.com/javase/8/docs/api/java/lang/Math.html#addExact-int-int-
https://docs.oracle.com/javase/7/docs/api/java/lang/Integer.html#MAX_VALUE

@Vampire ,
Can we use BigInteger and standard arithmetic operation on them ? https://docs.oracle.com/javase/7/docs/api/java/math/BigInteger.html#add(java.math.BigInteger)

Reply:
Well, I expected not to have more than Integer.MAX_VALUE findings and thought in case it does happen an exception is better than an overflow to MIN_VALUE. But we can of course use BigInteger.
done